app/helpers/dropdown/group_edit.rb

```diff
@@ -24,8 +24,11 @@ def initialize(template, group)
     private
 
     def init_items # rubocop:disable Metrics/AbcSize,Metrics/MethodLength
-      unless group.archived?
+      unless group.archived? || merge_candidates?
         add_item(translate(:merge), template.merge_group_path(group))
+      end
+
+      unless group.archived? || move_candidates?
         add_item(translate(:move), template.move_group_path(group))
       end
```

Comment:
I find complex `unless` constructs rather hard to read. Could you refactor it so that it gets more readable? E.g.:
```suggestion
      unless group.archived?
        if merge_candidates?
          add_item(translate(:merge), template.merge_group_path(group))
        end

        if move_candidates?
          add_item(translate(:move), template.move_group_path(group))
        end
      end
```
Actually, I think you misread it yourself and introduced a bug. Let me demonstrate:
```ruby
unless group.archived? || move_candidates?
# as per definition of "unless":
if !(group.archived? || move_candidates?)
# apply de Morgan's Theorem
if (!group.archived?) && (!move_candidates?)
```
Thus, in your implementation, the move option is only shown for non-archived groups which *do not have any move candidates*, which seems wrong to me.

The same goes for the move candidates.